### PR TITLE
Fix stop kill missing containers

### PR DESCRIPTION
**Pull request by zaineel**
## Type of Change
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update

## Motivation and Context
The `stop` and `kill` commands currently do not gracefully handle cases where container IDs are missing or not found. This PR adds proper error handling for missing containers and ensures validation tests run correctly.

Fixes #878

## Testing
- [x] Tested locally
- [x] Added/updated tests
- [ ] Added/updated docs

### CODE REVIEW

**Comment by zaineel** [comment]
@dcantah 
Could you please take a look at this PR? Thanks!

File: Sources/ContainerCommands/Container/ContainerStop.swift [hunk 1]
```diff
@@ -45,25 +52,20 @@ extension Application {
...
-                containers = try await ClientContainer.list()
-            } else {
-                containers = try await ClientContainer.list().filter { c in
-                    set.contains(c.id)
-                }
-            }
+            let allContainers = try await ClientContainer.list()
+            let containers = try self.all
+                ? allContainers
+                : Self.containers(matching: containerIds, in: allContainers)
```
Comment (Ronitsabhaya75) [inline Sources/ContainerCommands/Container/ContainerStop.swift:69]: I think we should handle ambiguous prefix matches here.

Currently, `allContainers.first(where:)` will silently pick the first matching container. If I have containers `abc` and `abd`, running `stop ab` might stop `abc` just because it appears first in the list.

It would be safer to filter all matches and throw an error if `matches.count > 1` (unless there's an exact match). This aligns with how Docker handles ambiguous prefixes.

File: Tests/ContainerCommandsTests/ContainerStopValidationTests.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+//===----------------------------------------------------------------------===//
+
+import ContainerClient
+import ContainerCommands
+import ContainerizationError
+import Testing
+
+@testable import ContainerCommands
+
+struct ContainerStopValidationTests {
```
Comment (Ronitsabhaya75) [inline Tests/ContainerCommandsTests/ContainerStopValidationTests.swift:24]: Could we add a test case for ambiguous prefixes?

We should verify that stop ab throws an error if both `abc` and `abd` exist. It's an important edge case to prevent accidental stops.

File: Sources/ContainerCommands/Container/ContainerStop.swift [hunk 3]
```diff
@@ -21,6 +21,13 @@ import ContainerizationError
 import ContainerizationOS
 import Foundation
 
+package protocol ContainerIdentifiable {
```
Comment (Ronitsabhaya75) [inline Sources/ContainerCommands/Container/ContainerStop.swift:25]: Refactoring this resolution logic into a shared helper is a great move. It keeps Stop and Kill consistent and makes testing way easier.

**Comment by zaineel** [comment]
@Ronitsabhaya75 Added ambiguous-prefix handling with exact-match preference, plus tests for
  ambiguity and exact IDs. Ready for re‑review.

**Comment by dcantah** [comment]
I think we can close this out as https://github.com/apple/container/pull/878 was merged. Thanks for trying to move this forward!
